server/test/services/documents/onDidChangeWatchedFiles.ts

```diff
@@ -1,218 +1,15 @@
-import { NoProject } from "@analyzer/NoProject";
-import { ClientTrackingState, ISolFileEntry } from "@common/types";
-import { onDidChangeWatchedFiles } from "@services/documents/onDidChangeWatchedFiles";
-import { assert } from "chai";
 import sinon from "sinon";
-import { FileChangeType } from "vscode-languageserver-protocol";
 import { ServerState } from "../../../src/types";
 import { setupMockLogger } from "../../helpers/setupMockLogger";
 import { setupMockTelemetry } from "../../helpers/setupMockTelemetry";
 
 describe("On did change watched files", () => {
-  describe("change to hardhat config file", () => {
-    it("should restart the worker", async () => {
-      const mockWorkerProcess = {
-        restart: sinon.spy(),
-      };
-
-      const serverState = setupServerState(mockWorkerProcess);
-
-      const [response] = await onDidChangeWatchedFiles(serverState)({
-        changes: [{ type: 1, uri: "/projects/example/hardhat.config.ts" }],
-      });
-
-      assert.deepStrictEqual(response, true);
-      assert(mockWorkerProcess.restart.called);
-    });
-
-    it("should gracefully fail if no project for config file", async () => {
-      const serverState = setupServerState();
-      serverState.projects = {};
-
-      const [response] = await onDidChangeWatchedFiles(serverState)({
-        changes: [{ type: 1, uri: "/projects/js-example/hardhat.config.js" }],
-      });
-
-      assert.deepStrictEqual(response, false);
-    });
-
-    it("should gracefully fail if no worker process for config file", async () => {
-      const serverState = setupServerState();
-      serverState.workerProcesses = {};
-
-      const [response] = await onDidChangeWatchedFiles(serverState)({
-        changes: [{ type: 1, uri: "/projects/example/hardhat.config.ts" }],
-      });
-
-      assert.deepStrictEqual(response, false);
-    });
-
-    it("should gracefully fail on an unexpected exception", async () => {
-      const serverState = setupServerState();
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      serverState.projects["/projects/example"] = undefined as any;
-
-      const [response] = await onDidChangeWatchedFiles(serverState)({
-        changes: [{ type: 1, uri: "/projects/example/hardhat.config.ts" }],
-      });
-
-      assert.deepStrictEqual(response, false);
-    });
-  });
-
-  describe("change to solidity file", () => {
-    let exampleSolFileEntry: ISolFileEntry;
-
-    beforeEach(() => {
-      exampleSolFileEntry = {
-        project: {
-          type: "hardhat",
-          configPath: "/projects/example/hardhat.config.ts",
-          basePath: "/projects/example",
-        },
-        tracking: ClientTrackingState.UNTRACKED,
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      } as any;
-    });
-
-    describe("that is untracked", () => {
```

Comment:
I tried to not touch it, but looking through references on this attribute, I'm not sure why is it important. I'm tempted to remove it completely, but perhaps you know of the reason why it exists?

Reply:
It was added in so we could answer `getOpenDocumentsInProject`, when passing those for validation.

This is instead of querying the client to get the currently open editor documents.

Reply:
Probably we don't need that. On that function, it's using the `serverState.documents` collection, which has the open editor documents already. In any case, I wouldn't change it now since it's out of scope for this PR.

Reply:
Is my mental model correct, is `serverState.documents` all docs that have ever been opened in the editor?

My vague recollection was `getopenDocumentsInProject` was trying to limit what was passed over to the validation thread, by filtering to the currently open docs.

But that was really just an attempt to mirror the original implementation while removing the need to have a call to the front-end to and back to get open docs (which tended to timeout and fail).

Do we have a feel for how big the `serverState.documents` collection might get under heavy use in a long editor session?

Reply:
Just tested to be sure. `serverState.documents` only retains the current open documents in the editor. So I feel safe removing the `tracked` flag from the SolFileEntry type.

Reply:
Brilliant.